sdk/Sdk.Generators/Constants.cs

```diff
@@ -14,7 +14,7 @@ internal static class BuildProperties
         {
             internal const string MSBuildRootNamespace = "build_property.RootNamespace";
             internal const string GeneratedCodeNamespace = "build_property.FunctionsGeneratedCodeNamespace";
-            internal const string EnableSourceGen = "build_property.FunctionsEnableMetadataSourceGen";
+            internal const string EnableSourceGen = "build_property.FunctionsEnableWorkerIndexing";
```

Comment:
Previously we had both `FunctionsEnableWorkerIndexing` and `FunctionsEnableMetadataSourceGen` since it is possible to use worker-indexing but still create and read from the `functions.json` file.  I think we can use just one prop as we only really want source-generated metadata if worker-indexing is enabled, and this first iteration reflects that.

Reply:
Did we already ship with this value? If so, we would need to fill the gap in behavior. We can rename it sure, but in the targets we should map the old property to the new one if the new one is not set yet.

Reply:
we only have [official docs](https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide#performance-optimizations) for using `FunctionsEnableWorkerIndexing` prop.

Reply:
Official docs or not, if we shipped it in a non-preview package we shouldn't break it.